lib/processors/bundlers/manifestBundler.js

```diff
@@ -57,25 +57,65 @@ class I18nResourceList {
  * @returns {Promise<module:@ui5/fs.Resource[]>} Promise resolving with manifest bundle resources
  */
 module.exports = ({resources, options: {namespace, bundleName, propertiesExtension, descriptor}}) => {
-	function getDescriptorI18nInfo(manifest) {
+	function bundleNameToUrl(bundleName, appId) {
+		if (!bundleName.startsWith(appId)) {
+			return null;
+		}
+		const relativeBundleName = bundleName.substring(appId.length + 1);
+		return relativeBundleName.replace(/\./g, "/") + propertiesExtension;
+	}
+
+	function addDescriptorI18nInfos(descriptorI18nInfos, manifest) {
+		function addI18nInfo(i18nPath) {
+			if (i18nPath.startsWith("ui5:")) {
+				log.warn(`Using the ui5:// protocol for i18n bundles is currently not supported ('${i18nPath}' in ${manifest.path})`);
+				return;
+			}
+			descriptorI18nInfos.set(
+				posixPath.join(posixPath.dirname(manifest.path), posixPath.dirname(i18nPath)),
+				posixPath.basename(i18nPath, propertiesExtension)
+			);
+		}
+
 		const content = JSON.parse(manifest.content);
-		let i18nFullPath = content["sap.app"]["i18n"];
+		const appI18n = content["sap.app"]["i18n"];
+		let i18nFullPath;
 		// i18n section in sap.app can be either a string or an object with bundleUrl
-		if (typeof i18nFullPath === "object") {
-			i18nFullPath = i18nFullPath.bundleUrl;
-		}
-		if (!i18nFullPath) {
+		if (typeof appI18n === "object") {
+			if (appI18n.bundleUrl) {
+				i18nFullPath = appI18n.bundleUrl;
+			} else if (appI18n.bundleName) {
+				i18nFullPath = bundleNameToUrl(appI18n.bundleName, content["sap.app"]["id"]);
```

Comment:
>  `content["sap.app"]["id"]`

Or should we rather use the namespace that has been passed into the processor?

I'm not sure, since they *should* be equal. But in the end the `namespace` will be used in the ui5-fs so maybe it's *more* correct?

Reply:
Afaics there could be multiple manifest.json files, so we need to use the one from the manifest.

Reply:
Ah, yes right!